In [10]:

import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns


In [12]:

# Read the data. Visadataset.csv file has 12 cols

path=r"C:\Users\User\OneDrive\Desktop\Omkar sir\DATA files\Visadataset.csv"


In [14]:


visa_df=pd.read_csv(r"C:\Users\User\OneDrive\Desktop\Omkar sir\DATA files\Visadataset.csv")
visa_df.head()


case_id continent education_of_employee has_job_experience  \
0  EZYV01      Asia           High School                  N   
1  EZYV02      Asia              Master's                  Y   
2  EZYV03      Asia            Bachelor's                  N   
3  EZYV04      Asia            Bachelor's                  N   
4  EZYV05    Africa              Master's                  Y   

  requires_job_training  no_of_employees  yr_of_estab region_of_employment  \
0                     N            14513         2007                 West   
1                     N             2412         2002            Northeast   
2                     Y            44444         2008                 West   
3                     N               98         1897                 West   
4                     N             1082         2005                South   

   prevailing_wage unit_of_wage full_time_position case_status  
0         592.2029         Hour                  Y      Denied  
1       83425.6500         Year                  Y   Certified  
2      122996.8600         Year                  Y      Denied  
3       83434.0300         Year                  Y      Denied  
4      149907.3900         Year                  Y   Certified

**PERCENTILE**

In [16]:

visa_df=pd.read_csv(r"C:\Users\User\OneDrive\Desktop\Omkar sir\DATA files\Visadataset.csv")
visa_df


case_id continent education_of_employee has_job_experience  \
0         EZYV01      Asia           High School                  N   
1         EZYV02      Asia              Master's                  Y   
2         EZYV03      Asia            Bachelor's                  N   
3         EZYV04      Asia            Bachelor's                  N   
4         EZYV05    Africa              Master's                  Y   
...          ...       ...                   ...                ...   
25475  EZYV25476      Asia            Bachelor's                  Y   
25476  EZYV25477      Asia           High School                  Y   
25477  EZYV25478      Asia              Master's                  Y   
25478  EZYV25479      Asia              Master's                  Y   
25479  EZYV25480      Asia            Bachelor's                  Y   

      requires_job_training  no_of_employees  yr_of_estab  \
0                         N            14513         2007   
1                         N             2412         2002   
2                         Y            44444         2008   
3                         N               98         1897   
4                         N             1082         2005   
...                     ...              ...          ...   
25475                     Y             2601         2008   
25476                     N             3274         2006   
25477                     N             1121         1910   
25478                     Y             1918         1887   
25479                     N             3195         1960   

      region_of_employment  prevailing_wage unit_of_wage full_time_position  \
0                     West         592.2029         Hour                  Y   
1                Northeast       83425.6500         Year                  Y   
2                     West      122996.8600         Year                  Y   
3                     West       83434.0300         Year                  Y   
4                    South      149907.3900         Year                  Y   
...                    ...              ...          ...                ...   
25475                South       77092.5700         Year                  Y   
25476            Northeast      279174.7900         Year                  Y   
25477                South      146298.8500         Year                  N   
25478                 West       86154.7700         Year                  Y   
25479              Midwest       70876.9100         Year                  Y   

      case_status  
0          Denied  
1       Certified  
2          Denied  
3          Denied  
4       Certified  
...           ...  
25475   Certified  
25476   Certified  
25477   Certified  
25478   Certified  
25479   Certified  

[25480 rows x 12 columns]

In [22]:
cat_columns=visa_df.select_dtypes(include='object').columns
num_columns=visa_df.select_dtypes(exclude='object').columns

num_columns

Index(['no_of_employees', 'yr_of_estab', 'prevailing_wage'], dtype='object')

In [138]:
df1=pd.DataFrame()
df1                                        # empty dataframe       

Empty DataFrame
Columns: []
Index: []

In [192]:

# USING FOR LOOP  with pandas and numpy built in methods TO GET SAME OUTPUT AS visa_df.describe() 


df1=pd.DataFrame()                                # empty dataframe

for i in num_columns:
    
    Count=len(visa_df[i])
    
    Min=min(visa_df[i])
    
    Max=max(visa_df[i])
    
    Mean=round(visa_df[i].mean(),2)
    
    Median=round(visa_df[i].median(),2)
    
    Std=round(visa_df[i].std(),2)
    
    p_25=round(np.percentile(visa_df[i],25),2)
    
    p_50=round(np.percentile(visa_df[i],50),2)
    
    p_75=round(np.percentile(visa_df[i],75),2)
    
    values=[Count,Min,Max,
    Mean,Median,Std,p_25,p_50,p_75]
    
    index=['count','min','max','mean','median','std','25p','50p','75p']
    
    cols=[i]
    
    df2=pd.DataFrame(values,index=index,columns=cols)
    
    df1=pd.concat([df1,df2],axis=1)                    # axis=0 means horizontal axis, axis =1 means vertical axis
    
    


In [194]:
  df1

no_of_employees  yr_of_estab  prevailing_wage
count          25480.00     25480.00       25480.0000
min              -26.00      1800.00           2.1367
max           602069.00      2016.00      319210.2700
mean            5667.04      1979.41       74455.8100
median          2109.00      1997.00       70308.2100
std            22877.93        42.37       52815.9400
25p             1022.00      1976.00       34015.4800
50p             2109.00      1997.00       70308.2100
75p             3504.00      2005.00      107735.5100

In [184]:
visa_df.describe()               # GIVES SAME OUTPUT AS ABOVE

no_of_employees   yr_of_estab  prevailing_wage
count     25480.000000  25480.000000     25480.000000
mean       5667.043210   1979.409929     74455.814592
std       22877.928848     42.366929     52815.942327
min         -26.000000   1800.000000         2.136700
25%        1022.000000   1976.000000     34015.480000
50%        2109.000000   1997.000000     70308.210000
75%        3504.000000   2005.000000    107735.512500
max      602069.000000   2016.000000    319210.270000

**Percentiles**
# 25p:
# Assume that 25p value = x
# 25percentage of data less than value : x
# For example prevailing_Wage 25p value is : 34015.47
# 25 percentage of people having salary less than 34015.47
# Total members : 25480
# 25percentage of 25480=6370
# which means 6370 people has salary less than 34015.47



In [108]:
# using Numpy

np.percentile(visa_df['prevailing_wage'],25)                     # number of ppl with prevailing wage < 25 percent


34015.479999999996

In [33]:

25*25480/100                                     # 25% OF total employees


6370.0

In [37]:
# step-1: Read the prevailing wage
# STEP 2: PROVODE THE PREVAILING CONDITION
# prevailing wage < percentile value

# Using percentile() method from Numpy

wage_data=visa_df['prevailing_wage']

p_25=np.percentile(visa_df['prevailing_wage'],25)                         # number of ppl with prevailing wage < 25 percent

con=wage_data<p_25                                         # since using reltional operator < , output is true or false ONLY

con                                                         # true means prevailing wage < percentile value (ie 6370)
                                                           # false means prevailing wage > percentile value (ie 6370)...chk from main file


0         True
1        False
2        False
3        False
4        False
         ...  
25475    False
25476    False
25477    False
25478    False
25479    False
Name: prevailing_wage, Length: 25480, dtype: bool

In [39]:
len(visa_df[con])                                            # number of ppl having < 6370

6370



**50p:**
- Assume that 50p value = x
- 50percentage of data less than value : x
- For example prevailing_Wage 50p value is : 70308
- 50 percentage of people having salary less than 70308
- Total members : 25480
- 50 percentage of 25480= 12740
- which means 12740 people has salary less than 70308






In [46]:
# 50 percentile mark of prevailing wage using dataframe and np percentile()

wage_data=visa_df['prevailing_wage']

p_50=np.percentile(visa_df['prevailing_wage'],50)

con=wage_data<p_50

len(visa_df[con])                                       # number of ppl with prevailing wage < 50 percent


12740

In [48]:
# 75 percentile mark of previaling wage

wage_data=visa_df['prevailing_wage']

p_75=np.percentile(visa_df['prevailing_wage'],75)

con=wage_data<p_75

len(visa_df[con])                                 # number of ppl with prevailing wage < 75 percent


19110

In [116]:
# 100 percentile mark of previaling wage

wage_data=visa_df['prevailing_wage']

p_100=np.percentile(visa_df['prevailing_wage'],100)

con=wage_data<p_100

len(visa_df[con])                                 # number of ppl with prevailing wage < 100 percent


25479

**Emperical Rule**

- (mean-1 std) to (mean+1std) : 68%
- (mean-2 std) to (mean+2std) : 95%
- (mean-3 std) to (mean+2std) : 98%

In [ ]:
# step-1: calculate the mean of prevailing wage

# step-2: calculate the std of prevailig wage

# step-3: lb=mean-1*std                                      lb is lower bound                         

# step-4: ub=mean+1*std                                      ub is upper bound

# step-5: con1= data<ub                                     condition 1

# step-6: con2= data>lb                                     condition 2

# step-7: con3= con1 and con2                               condition 3

# step-8: data[con3]

# step-9: compare with 68% 25480 count


In [104]:
# PART 1: to find the prevailing wage between lb and ub values: 

mu = visa_df['prevailing_wage'].mean()

print("\n\n")

print(" the mean of prevailing wage is =",round(mu,2))

sigma = visa_df['prevailing_wage'].std()

print("\n the std deviation of prevailing wage is = ",round(sigma,2))

lb = mu-1*sigma                                                                     # formula for lb

print(" \nthe lower bound value is = ",round(lb,2))

ub = mu+1*sigma                                                                     # formula for lb

print("\n the upper bound value is =",round(ub,2))

print("\n\n")

##########################################################################################






 the mean of prevailing wage is = 74455.81

 the std deviation of prevailing wage is =  52815.94
 
the lower bound value is =  21639.87

 the upper bound value is = 127271.76





In [106]:
# PART 2: prevailing wage between lb and ub values of 21639 and 127271

con1=visa_df['prevailing_wage'] > lb

con2=visa_df['prevailing_wage'] < ub

con3= con1 & con2                               # condition 1 and 2 is True

visa_df[con3]


case_id continent education_of_employee has_job_experience  \
1         EZYV02      Asia              Master's                  Y   
2         EZYV03      Asia            Bachelor's                  N   
3         EZYV04      Asia            Bachelor's                  N   
5         EZYV06      Asia              Master's                  Y   
6         EZYV07      Asia            Bachelor's                  N   
...          ...       ...                   ...                ...   
25473  EZYV25474      Asia            Bachelor's                  Y   
25474  EZYV25475    Africa             Doctorate                  N   
25475  EZYV25476      Asia            Bachelor's                  Y   
25478  EZYV25479      Asia              Master's                  Y   
25479  EZYV25480      Asia            Bachelor's                  Y   

      requires_job_training  no_of_employees  yr_of_estab  \
1                         N             2412         2002   
2                         Y            44444         2008   
3                         N               98         1897   
5                         N             2339         2012   
6                         N             4985         1994   
...                     ...              ...          ...   
25473                     N             2421         2007   
25474                     N             2594         1979   
25475                     Y             2601         2008   
25478                     Y             1918         1887   
25479                     N             3195         1960   

      region_of_employment  prevailing_wage unit_of_wage full_time_position  \
1                Northeast         83425.65         Year                  Y   
2                     West        122996.86         Year                  Y   
3                     West         83434.03         Year                  Y   
5                    South         78252.14         Year                  Y   
6                    South         53635.39         Year                  Y   
...                    ...              ...          ...                ...   
25473            Northeast         22845.56         Year                  Y   
25474            Northeast         51104.78         Year                  Y   
25475                South         77092.57         Year                  Y   
25478                 West         86154.77         Year                  Y   
25479              Midwest         70876.91         Year                  Y   

      case_status  
1       Certified  
2          Denied  
3          Denied  
5       Certified  
6       Certified  
...           ...  
25473   Certified  
25474   Certified  
25475   Certified  
25478   Certified  
25479   Certified  

[17171 rows x 12 columns]